### ISSUE TITLE
Error during update from ver 4.4 to 5

### ISSUE THREAD

**Issue by Andrew-W-Robinson** [YetiForceCompany/YetiForceCRM#9660]
<br><hr>Generated by YetiforceCRM ver. 4.4.0<br>PHP: 7.1.17<br><br><strong>Server configuration errors:</strong><br>session.gc_maxlifetime: 1440<br>auto_detect_line_endings: Off<br><strong>Library errors:</strong><br>exif_read_data<br>ldap_connect<br>opcache_get_configuration<br>apcu<br><br>I am getting an error when updating to version 5:-<br>(All Cron are disabled)<br>Error!!!<br><br>SQLSTATE[42S02]: Base table or view not found: 1146 Table 'buco_crmtest.u_yf_chat_messages' doesn't exist Failed to prepare SQL: SELECT * FROM `u_yf_chat_messages` LIMIT 20<br><br>Go backHome page<br><br>There is an existing table before the update, but it is deleted during the update.<br><br>Any help or hints would be appreciated.<br>

**Comment by TomaszQr** [comment]
Please attach the system logs from the update, this SELECT * FROM `u_yf_chat_messages` LIMIT 20 query isn't triggered during the system update.

**Comment by vovpff** [comment]
@TomaszQr I think this query triggered after update when page loading.

**Comment by Andrew-W-Robinson** [comment]
Hi TomaszQr,

 

I have run the update three times with V1 update and twice with V2 update with the same result, the 4.4 version was fine before the update.

 

Please find the logs attached if you need different logs, just ask.

 

Thanks

**Comment by alagmouch** [comment]
I encountered the same issue. Fortunately for me, I had a backup. So I rolled back the database and the directory to 4.4.

I tried updating several times, with an update package from SourceForge as well from GitHub, and each time rolling back to 4.4. (I also disabled cron during the update.)

Comparing the database of 4.4 with the db 5.0 updated version, it shows that the table **u_yf_chat_messages** does not exist anymore in db 5.0, while in db 4.4 the table is still present.

The old chat module is referring to it, it seems. So after updating to 5.0, it hangs each time the home screen is initiated. Although, if you clear the browser cache and open the application again, the login screen shows as normal. And once you login, the error comes again and you are stuck again.

So I tried turning off the chat by default prior to upgrading. Did this in **config/config_override.php** with variable `$CHAT_DISPLAY = 'false'; ` but this does not hide the Chat button in the upper menu. So the variable didn't work. May I was doing something wrong.

And if you turn off the Chat Module in System Admin in 4.4, the access to the module is gone, but the Chat button still loads and wants to initiate and gives errors in **cache/logs/system.log**. Although the errors are there in system.log, the CRM works.

I'm currently still on 4.4. I was actually waiting for version 5.1 and test again, and see what the outcome is before posting anything. But now that someone else is encountering it also, I think it's good to share my part of the quarter also.

Hopefully, you can use this info.

**System Info**
`Nginx 1.15.8`
`PHP 7.2.14`
`Zend Engine v3.2.0`
`Zend OPcache v7.2.14`
`10.1.37 MariaDB Server`

**Comment by Andrew-W-Robinson** [comment]
I have tried the newer V3 update with the same error result and subsequent update failure. V2 logs attached.
[updateLogsTrace.log](https://github.com/YetiForceCompany/YetiForceCRM/files/2936613/updateLogsTrace.log)
[Importer.log](https://github.com/YetiForceCompany/YetiForceCRM/files/2936642/Importer.log)

**Comment by mariuszkrzaczkowski** [comment]
did you install version 4.4.0 or did you update ??
the logs show an incorrectly updated system from 4.3.0 to 4.4.0

**Comment by Andrew-W-Robinson** [comment]
I tried to update 4.4 to 5.0, you must have some old logs. Here are the logs from my latest attempt with V3 update.
[Importer.log](https://github.com/YetiForceCompany/YetiForceCRM/files/2942161/Importer.log)
[updateLogsTrace.log](https://github.com/YetiForceCompany/YetiForceCRM/files/2942162/updateLogsTrace.log)

**Comment by alagmouch** [comment]
As a quick update. For whoever wants to know. I used v3 this time to update from 4.4.0 to 5.0. It went well. Update went through. No issues. 4.4.0 was a fresh install prior to updating to 5.0. No error anymore at my side.

**Comment by mariuszkrzaczkowski** [comment]
@Andrew-W-Robinson 
answer to my previous questions are important

**Comment by Andrew-W-Robinson** [comment]
Mariusz I am updating version 4.4 to 5.0 using version 3 of the update file. Unless the logs I have sent on my last comment are not the ones you require, they do refer to the most recent attempt to update, I can even see the error message at the end of log. Unfortunately I do not have the luxury of being able to use a fresh install to cure the issue.

**Comment by mariuszkrzaczkowski** [comment]
did you install the clean version 4.4.0 or did you update from 4.3.0 to 4.4.0? 
in the update logs (4.4.0 to 5.0.0) you can see the old table structure from version 4.3.0.
This may be due to an incorrect update from 4.3.0 to 4.4.0

this error has already been reported in https://github.com/YetiForceCompany/YetiForceCRM/issues/9504

**Comment by Andrew-W-Robinson** [comment]
I updated the version from previous versions 4.3.0 to 4.4.0 and then the latest update. I referred the issue to my server admin who have run the database queries you suggest in #9504, they say no errors occurred and it completed successfully but have not cured the issue and do not believe it made any difference. Then to quote 
'I initially ran the update and it failed stating "Incorrect request Illegal value in field phone", at this point the update had stopped due to this error and upon attempting to access the site it provides the previously seen error - "SQLSTATE[42S02]: Base table or view not found: 1146 Table 'buco_crmtest.u_yf_chat_messages' doesn't exist Failed to prepare SQL: SELECT * FROM u_yf_chat_messages LIMIT 20". This error however is being generated because the update hasn't completed and is yet to create the necessary tables. I then removed the number from the "phone field" and re-ran the update, this time it failed however the illegal field was the fax number, which again I removed. Following the removal of both the phone and fax field the update then completed successfully. Based on this there was never an issue with the update itself, it no doubt failed every single time it was attempted previously due to the two aforementioned fields and therefore the error was then presented as the update was incomplete.' 
I can now see a working copy of the updated Version 5.0.0. I hope this helps anyone who may be experiencing the same issue. it looks like the same procedure will have to be done when I finally attempt to update my working copy of version 4.4.0 before it will update correctly.

**Comment by Andrew-W-Robinson** [comment]
Further to my comments above it was specifically the phone and fax fields in the Company details Software Configuration > Company Information > Company Details. After the relatively successful update the details pages are now blank although the summary pages are still working correctly.

**Comment by alagmouch** [comment]
@Andrew-W-Robinson I had the same issue also. Number invalid, which was another cause of the update to not correctly go through. What helped in our case, prior to applying the update, is to create a **config_override.php** file in the **config/** directory and set the following value to false. (Especially if you upload Leads as bulk, you will encounter incorrect numbers so now and then.)

`// Enable advanced phone number validation. Enabling  it will block saving invalid phone number.
$phoneFieldAdvancedVerification = false;`

The strict validation of the phone numbers is turned off then, I assumed.

**Comment by Andrew-W-Robinson** [comment]
My issue was caused by the tel and fax numbers fields in the company details, which have been deleted in the version 5.0.0. and apparently not due to validation. I still cannot update my working copy as all the details pages are blank. Any fix for this? I am waiting for version 5.1.0 to see if it fixed in this version.

**Comment by alagmouch** [comment]
I'm sorry. I can't help you with this. I didn't encounter that. Although, opening the Activities tab in a contact and creating an activity (whether from the summary widget or the activities tab) is very, very slow. I rolled back to wait for 5.1 to see whether that is still the case, before I register an Issue for it. The Extended view was added. I though it was related to the view. Changed it back to Standard in the config, but that didn't help. All the rest is quite fast still.
